azure.js

```diff
@@ -44,37 +46,117 @@ module.exports = (args, sendTo) => {
   }
 
   const processVirtualMachineScaleSet = function (client, ips, localIps, vmssName) {
-    client.networkInterfaces._listVirtualMachineScaleSetNetworkInterfaces(args.azure, vmssName, function(err, results) {
-      if (err) {
-        console.log(err);
-        return;
-      }
-
-      results.forEach(result => {
-        result.ipConfigurations.forEach(ip => {
-          if (!localIps.includes(ip.privateIPAddress)) {
-            ips.push(ip.privateIPAddress)
+    client.networkInterfaces
+      ._listVirtualMachineScaleSetNetworkInterfaces(args.azure, vmssName,
+        function (err, results) {
+          if (err) {
+            console.log(err);
+            return;
           }
+
+          results.forEach(result => {
+            result.ipConfigurations.forEach(ip => {
+              if (!localIps.includes(ip.privateIPAddress)) {
+                ips.push(ip.privateIPAddress)
+              }
+            })
+          })
+
+          updateSendTo(ips)
         })
-      })
+  }
 
-      updateSendTo(ips)
+  const processVirtualMachineScaleSetPromise = function (
+    client,
+    vmssName
+  ) {
+    return new Promise((resolve, reject) => {
+      client.networkInterfaces._listVirtualMachineScaleSetNetworkInterfaces(
+        args.azure,
+        vmssName,
+        function (err, results) {
+          if (err) {
+            // Do nothing if the VMSS is deleted, so that for next deployment inter VMSS forwarding will work
```

Comment:
Is this still needed when using tags?

Reply:
Here we are just reducing the error logs even with tags we may get this error once during the vmss delete.